From 4f3c8b8d692eb50947d891fa9666b444bce35899 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Tue, 12 Feb 2019 13:04:55 +0100
Subject: [PATCH 121/152] workspace: Continue overlay idle hide timeout when
 pointing at close button

While it makes sense to remove the timeout in every other case, restart it
if the pointer is hovering over the close button to make sure the overlay
will be hidden afterwards.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/683
---
 js/ui/workspace.js | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/js/ui/workspace.js b/js/ui/workspace.js
index f09637b89..72707173c 100644
--- a/js/ui/workspace.js
+++ b/js/ui/workspace.js
@@ -693,12 +693,13 @@ var WindowOverlay = class {
     }
 
     _idleHideOverlay() {
-        this._idleHideOverlayId = 0;
+        if (this.closeButton['has-pointer'])
+            return GLib.SOURCE_CONTINUE;
 
-        if (!this._windowClone['has-pointer'] &&
-            !this.closeButton['has-pointer'])
+        if (!this._windowClone['has-pointer'])
             this._animateInvisible();
 
+        this._idleHideOverlayId = 0;
         return GLib.SOURCE_REMOVE;
     }
 
-- 
2.23.0

